### ISSUE TITLE
GFM indented fenced code block fails to exit for some inner parsers

### ISSUE THREAD

**Issue by lishid** [codemirror/CodeMirror#6884]
Open https://codemirror.net/mode/gfm/index.html

Run the following code in console to enable `shell` language.
```js
let s = document.createElement('script');
s.src='../shell/shell.js';
document.head.appendChild(s);
```

Type the following into the editor:
````markdown
- list
	```shell
	cat <<
	```

Some more text
````

Expected: Some more text should not be highlighted
Actual:
![image](https://user-images.githubusercontent.com/609710/155017122-793b8d94-7d66-4007-9fff-e7f2fe2981bf.png)

**Comment by marijnh** [comment]
Weren't you using CM6? The Markdown stuff should definitely be a lot less broken there. The CM5 Markdown/GFM stuff is definitely in low-maintenance mode (it's contributed code, and not in great shape).

**Comment by lishid** [comment]
> Weren't you using CM6?

That is correct, we are in the middle of an awkward migration phase from CM5 to CM6 where we currently still offer CM5 as a legacy editor, and uses a proxy object to expose a common interface for interacting with both CM5 and CM6.

For parsing we are, unfortunately, still stuck on the old CM5 parsers through `stream-parser` until we can fully remove CM5 from the codebase after which we would be able to commit a big chunk of development time to rewrite our markdown flavor in CM6's lezer.

> The CM5 Markdown/GFM stuff is definitely in low-maintenance mode (it's contributed code, and not in great shape).

Ok fair enough, that's understandable. I will see if I can find some time to fix this one from our end.
